src/segment-loader.js

```diff
@@ -717,6 +717,94 @@ export default class SegmentLoader extends videojs.EventTarget {
     };
   }
 
+  abortRequestEarly_(stats) {
+    // If the player is not paused and the current playlist is not the
+    // lowestEnabledRendition, consider the possibility of aborting the current request
+    // early for an emergency downswitch
+    // TODO: Replace timeout with a boolean indicating whether this playlist is the
+    //       lowestEnabledRendition
+    if (!this.hls_.tech_.paused() && this.xhrOptions_.timeout) {
+      const firstByteReceived = Date.now() - stats.firstByteReceived;
+
+      // Wait at least 1 second before using the calculated bandwidth from the
+      // progress event to allow the bitrate to stableize
+      if (firstByteReceived > 1000) {
+        const realBandwidth = stats.bandwidth;
+        const estimatedSize =
+          this.pendingSegment_.duration * this.playlist_.attributes.BANDWIDTH;
```

Comment:
We should check for attributes and attributes.BANDWIDTH higher up and return if not available.